Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -997,16 +996,15 @@ - (void)applyReferralCode:(NSString *)code andCallback:(callbackWithParams)callb
 
 #pragma mark - Private methods
 
-+ (Branch *)getBranchInstance:(BOOL)isLive {
-    if (!currInstance) {
-        NSString *branchKey = [BNCPreferenceHelper getBranchKey:isLive];
-        if (!branchKey || [branchKey isEqualToString:NO_STRING_VALUE]) {
-            NSLog(@"Branch Warning: Please enter your branch_key in the plist!");
-        }
-        
-        [Branch initInstance];
-    }
-    return currInstance;
++ (Branch *)getInstanceInternal {
+    static Branch *branch;
```

Comment:
If it's a static variable scoped to the method, won't this be garbage collected if we're not created in the AppDelegate? Let's say someone inits Branch in VC1, then navigates a few VCs down the funnel causing VC1 to be GCed. My understanding is that a static variable in the class will remain in memory but a method variable could be GC. Unsure here though.